x-pack/plugins/security_solution/server/lib/detection_engine/rule_execution_log/dev_tools_events.txt

```diff
@@ -0,0 +1,344 @@
+  "rule": {
+    "id": "1234-56789-dfgdfhgfgh-122346567"
+  },
+  "kibana": {
+    "saved_objects": [
+      {
+        "type": "alert",
+        "id": "1234-56789-dfgdfhgfgh-122346567",
+        "namespace": "some-space"
+      }
+    ]
+  }
+}
+
+POST /try-ecs-event-log/_doc
+{
+  "ecs": {
+    "version": "1.9.0"
+  },
+  "event": {
+    "provider": "detection-engine",
+    "dataset": "detection-engine.rule-execution-log",
+    "severity": 10,
+    "kind": "metric",
+    "type": [
+      "info"
+    ],
+    "action": "metric-indexing-duration-sum",
+    "duration": 13070000,
+    "created": "2021-03-09T10:55:54.568Z",
+    "sequence": 6
+  },
+  "log": {
+    "logger": "detection-engine.rule-execution-log",
+    "level": "info"
+  },
+  "@timestamp": "2021-03-09T10:55:54.567Z",
+  "message": "Missing read privileges on indices [your-index-1, your-index-2]",
+  "rule": {
+    "id": "1234-56789-dfgdfhgfgh-122346567"
+  },
+  "kibana": {
+    "saved_objects": [
+      {
+        "type": "alert",
+        "id": "1234-56789-dfgdfhgfgh-122346567",
+        "namespace": "some-space"
+      }
+    ]
+  }
+}
+
+POST /try-ecs-event-log/_doc
+{
+  "ecs": {
+    "version": "1.9.0"
+  },
+  "event": {
+    "provider": "detection-engine",
+    "dataset": "detection-engine.rule-execution-log",
+    "severity": 10,
+    "kind": "metric",
+    "type": [
+      "info"
+    ],
+    "action": "metric-indexing-lookback",
+    "end": "2021-03-09T10:55:54.567Z",
+    "created": "2021-03-09T10:55:54.568Z",
+    "sequence": 7
+  },
+  "log": {
+    "logger": "detection-engine.rule-execution-log",
+    "level": "info"
+  },
+  "@timestamp": "2021-03-09T10:55:54.567Z",
+  "message": "Missing read privileges on indices [your-index-1, your-index-2]",
+  "rule": {
+    "id": "1234-56789-dfgdfhgfgh-122346567"
+  },
+  "kibana": {
+    "saved_objects": [
+      {
+        "type": "alert",
+        "id": "1234-56789-dfgdfhgfgh-122346567",
+        "namespace": "some-space"
+      }
+    ]
+  }
+}
+
+POST /try-ecs-event-log/_doc
+{
+  "ecs": {
+    "version": "1.9.0"
+  },
+  "event": {
+    "provider": "detection-engine",
+    "dataset": "detection-engine.rule-execution-log",
+    "severity": 20,
+    "kind": "event",
+    "type": [
+      "change"
+    ],
+    "action": "status-changed",
+    "created": "2021-03-09T10:55:54.568Z",
+    "sequence": 8
+  },
+  "log": {
+    "logger": "detection-engine.rule-execution-log",
+    "level": "warning"
+  },
+  "@timestamp": "2021-03-09T10:55:54.567Z",
+  "message": "Missing read privileges on indices [your-index-1, your-index-2]",
+  "rule": {
+    "id": "1234-56789-dfgdfhgfgh-122346567"
+  },
+  "kibana": {
+    "saved_objects": [
+      {
+        "type": "alert",
+        "id": "1234-56789-dfgdfhgfgh-122346567",
+        "namespace": "some-space"
+      }
+    ],
+    "detection_engine": {
+      "rule_status": "warning",
+      "rule_status_severity": 20
+    }
```

Comment:
Yeah, that's a good point... I was hoping that I would be able to map everything to native ECS fields.

`message` is mapped as text, not keyword. And we might need to have both status and message in a single event, e.g. if we decide to show status updates in the log. I mean, it would be simple and nice to just have the ability to have an additional message field.

`event.kind` has a strict set of allowed values, as long as we decide to conform to ECS, of course.

[`labels`](https://www.elastic.co/guide/en/ecs/current/ecs-base.html#field-labels) could be used for that, but this seems to me the same thing if we talk about event log. We'd either need to specify labels mapping in event log (same solution as introducing a custom field set), or turn on dynamic mapping for `labels`. I'm not sure about the latter since event log is designed as a single multitenant index. If it could support separate indices per each tenant/provider, that would be a different story. @pmuellr what do you think about these tradeoffs?

As for other native ECS fields, I just didn't find any suitable.

Reply:
Ah okay I didn't realize event.kind had limited values.. Maybe we could utilize [`event.code`](https://www.elastic.co/guide/en/ecs/current/ecs-event.html#field-event-code)? That seems to be a flexible field? I"m just trying to think of ways to remove the detection engine specific fields.

Reply:
Idk if it would cause any issues here but i know we use the `event.code` field for a lot of endpoint exception stuff